pkg/model/sarif.go

```diff
@@ -0,0 +1,289 @@
+	queryDescription string
+	queryURI         string
+	queryCategory    string
+	severity         Severity
+}
+
+type sarifMessage struct {
+	Text string `json:"text"`
+}
+
+type sarifComponentReference struct {
+	ComponentReferenceName  string `json:"name"`
+	ComponentReferenceGUID  string `json:"guid"`
+	ComponentReferenceIndex int    `json:"index"`
+}
+
+type sarifDescriptorReference struct {
+	ReferenceID    string                  `json:"id"`
+	ReferenceIndex int                     `json:"index"`
+	ToolComponent  sarifComponentReference `json:"toolComponent"`
+}
+
+type sarifDescriptorRelationship struct {
+	Target sarifDescriptorReference `json:"target"`
+}
+
+type sarifConfiguration struct {
+	Level string `json:"level"`
+}
+
+type sarifRule struct {
+	RuleID               string                        `json:"id"`
+	RuleName             string                        `json:"name"`
+	RuleShortDescription sarifMessage                  `json:"shortDescription"`
+	RuleFullDescription  sarifMessage                  `json:"fullDescription"`
+	DefaultConfiguration sarifConfiguration            `json:"defaultConfiguration"`
+	HelpURI              string                        `json:"helpUri"`
+	RuleRelationships    []sarifDescriptorRelationship `json:"relationships"`
+}
+
+type sarifDriver struct {
+	ToolName     string      `json:"name"`
+	ToolVersion  string      `json:"version"`
+	ToolFullName string      `json:"fullName"`
+	ToolURI      string      `json:"informationUri"`
+	Rules        []sarifRule `json:"rules"`
+}
+
+type sarifTool struct {
+	Driver sarifDriver `json:"driver"`
+}
+
+type sarifRegion struct {
+	StartLine int `json:"startLine"`
+}
+
+type sarifArtifactLocation struct {
+	ArtifactURI string `json:"uri"`
+}
+
+type sarifPhysicalLocation struct {
+	ArtifactLocation sarifArtifactLocation `json:"artifactLocation"`
+	Region           sarifRegion           `json:"region"`
+}
+
+type sarifLocation struct {
+	PhysicalLocation sarifPhysicalLocation `json:"physicalLocation"`
+}
+
+type sarifResult struct {
+	ResultRuleID    string          `json:"ruleId"`
+	ResultRuleIndex int             `json:"ruleIndex"`
+	ResultKind      string          `json:"kind"`
+	ResultMessage   sarifMessage    `json:"message"`
+	ResultLocations []sarifLocation `json:"locations"`
+}
+
+type sarifTaxanomyDefinition struct {
+	DefinitionID               string       `json:"id"`
+	DefinitionName             string       `json:"name"`
+	DefinitionShortDescription sarifMessage `json:"shortDescription"`
+	DefinitionFullDescription  sarifMessage `json:"fullDescription"`
+}
+
+type sarifTaxonomy struct {
+	TaxonomyGUID             string                    `json:"guid"`
+	TaxonomyName             string                    `json:"name"`
+	TaxonomyFullDescription  sarifMessage              `json:"fullDescription"`
+	TaxonomyShortDescription sarifMessage              `json:"shortDescription"`
+	TaxonomyDefinitions      []sarifTaxanomyDefinition `json:"taxa"`
```

Comment:
why taxa name?

Reply:
This is a definition for taxonomies object as you can see [here](https://docs.oasis-open.org/sarif/sarif/v2.1.0/cs01/sarif-v2.1.0-cs01.html#_Toc16012492)